node/src/service.rs

```diff
@@ -41,158 +40,227 @@ pub fn build_inherent_data_providers(sr25519_public_key: sr25519::Public) -> Res
 	Ok(providers)
 }
 
-/// Starts a `ServiceBuilder` for a full service.
-///
-/// Use this macro if you don't actually need the full service, but just the builder in order to
-/// be able to perform chain operations.
-macro_rules! new_full_start {
-	($config:expr, $sr25519_public_key:expr) => {{
-		let mut import_setup = None;
-		let inherent_data_providers = crate::service::build_inherent_data_providers($sr25519_public_key)?;
-
-		let builder = sc_service::ServiceBuilder::new_full::<
-			utxo_runtime::opaque::Block, utxo_runtime::RuntimeApi, crate::service::Executor
-		>($config)?
-			.with_select_chain(|_config, backend| {
-				Ok(sc_consensus::LongestChain::new(backend.clone()))
-			})?
-			.with_transaction_pool(|config, client, _fetcher, prometheus_registry| {
-				let pool_api = sc_transaction_pool::FullChainApi::new(client.clone());
-				Ok(sc_transaction_pool::BasicPool::new(
-					config,
-					std::sync::Arc::new(pool_api),
-					prometheus_registry,
-				))
-			})?
-			.with_import_queue(
-				|_config, client, select_chain, _transaction_pool, spawn_task_handle, registry| {
-
-				let pow_block_import = sc_consensus_pow::PowBlockImport::new(
-					client.clone(),
-					client.clone(),
-					sha3pow::Sha3Algorithm::new(client.clone()),
-					0, // check inherents starting at block 0
-					select_chain,
-					inherent_data_providers.clone(),
-				);
-
-				let import_queue = sc_consensus_pow::import_queue(
-					Box::new(pow_block_import.clone()),
-					None,
-					None,
-					sha3pow::Sha3Algorithm::new(client.clone()),
-					inherent_data_providers.clone(),
-					spawn_task_handle,
-					registry,
-				)?;
-
-				import_setup = Some(pow_block_import);
-
-				Ok(import_queue)
-			})?;
-
-		(builder, import_setup, inherent_data_providers)
-	}}
+type FullClient = sc_service::TFullClient<Block, RuntimeApi, Executor>;
+type FullBackend = sc_service::TFullBackend<Block>;
+type FullSelectChain = sc_consensus::LongestChain<FullBackend, Block>;
+type PowBlockImport = sc_consensus_pow::PowBlockImport<Block, Arc<FullClient>, FullClient, FullSelectChain, Sha3Algorithm<FullClient>, CanAuthorWithNativeVersion<Executor>>;
+// HELP NEEDED! -->                                                                                 What type is supposed to go here?? ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
@expenses I need help with types again. My ultimate goal is to return the `pow_block_import` from `new_partial` to use it in `new_full`.

Am I doing something wrong to have to manually figure out these types?